### ISSUE TITLE
Training error under tf1.0

### ISSUE THREAD

**Issue by xray1111** [weinman/cnn_lstm_ctc_ocr#25]
Hi,

I cloned the project and try to train it under tensorflow 1.0, but got the following error information. Could you please give me some advice? Thank you very much! BTW, I'm using ubuntu 16.04 and IBM ppc64 machine.

```
xiaoren@S822lc1:~/homework/cnn_lstm_ctc_ocr/src$ python train.py
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcublas.so.8.0 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcudnn.so.5 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcufft.so.8.0 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcurand.so.8.0 locally
Traceback (most recent call last):
  File "train.py", line 207, in <module>
    tf.app.run()
  File "/opt/DL/tensorflow/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 44, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "train.py", line 173, in main
    image,width,label = _get_input()
  File "train.py", line 83, in _get_input
    length_threshold=FLAGS.length_threshold )
  File "/home/xiaoren/homework/cnn_lstm_ctc_ocr/src/mjsynth.py", line 69, in bucketed_input_pipeline
    dynamic_pad=True)
  File "/opt/DL/tensorflow/lib/python2.7/site-packages/tensorflow/contrib/training/python/training/bucket_ops.py", line 389, in bucket_by_sequence_length
    shared_name=shared_name)
  File "/opt/DL/tensorflow/lib/python2.7/site-packages/tensorflow/contrib/training/python/training/bucket_ops.py", line 231, in bucket
    control_flow_ops.no_op)
  File "/opt/DL/tensorflow/lib/python2.7/site-packages/tensorflow/python/ops/control_flow_ops.py", line 1726, in cond
    raise TypeError("pred must not be a Python bool")
TypeError: pred must not be a Python bool
```

**Comment by weinman** [comment]
I'd not encountered that before; I'm not sure why. My guess is that the root cause is the `keep_input` variable being assigned on line 137 of mjsynth.py. See if perhaps changing 
```keep_input = True```
to something like the following will fix it. If it does, I'll push the change to the master branch (or accept a pull request to that effect).
```keep_input = tf.constant(True, dtype=tf.bool)```

**Comment by weinman** [comment]
Fixed in  fc928d0.
